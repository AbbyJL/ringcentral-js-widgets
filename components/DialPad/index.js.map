{"version":3,"sources":["components/DialPad/index.js"],"names":["DialPad","keyConfig","value","text","dx","alternativeValue","props","root","className","map","row","rowIdx","btn","hideSpecial","btnPlaceholder","onButtonPress","onButtonOutput","alternativeTimeout","propTypes","string","bool","func","number"],"mappings":";;;;;kBAkBwBA,O;;AAlBxB;;;;AACA;;;;AACA;;;;AAEA;;;;AACA;;;;;;AAEA,IAAMC,YAAY,CAChB,CAAC,EAAEC,OAAO,GAAT,EAAcC,MAAM,EAApB,EAAD,EAA2B,EAAED,OAAO,GAAT,EAAcC,MAAM,KAApB,EAA2BC,IAAI,SAA/B,EAA3B,EAAuE,EAAEF,OAAO,GAAT,EAAcC,MAAM,KAApB,EAA2BC,IAAI,SAA/B,EAAvE,CADgB,EAEhB,CAAC,EAAEF,OAAO,GAAT,EAAcC,MAAM,KAApB,EAA2BC,IAAI,SAA/B,EAAD,EAA6C,EAAEF,OAAO,GAAT,EAAcC,MAAM,KAApB,EAA2BC,IAAI,UAA/B,EAA7C,EAA0F,EAAEF,OAAO,GAAT,EAAcC,MAAM,KAApB,EAA2BC,IAAI,SAA/B,EAA1F,CAFgB,EAGhB,CAAC,EAAEF,OAAO,GAAT,EAAcC,MAAM,MAApB,EAA4BC,IAAI,UAAhC,EAAD,EAA+C,EAAEF,OAAO,GAAT,EAAcC,MAAM,KAApB,EAA2BC,IAAI,SAA/B,EAA/C,EAA2F,EAAEF,OAAO,GAAT,EAAcC,MAAM,MAApB,EAA4BC,IAAI,UAAhC,EAA3F,CAHgB,EAIhB,CACE,EAAEF,OAAO,GAAT,EAAcC,MAAM,EAApB,EADF,EAEE,EAAED,OAAO,GAAT,EAAcC,MAAM,GAApB,EAAyBE,kBAAkB,GAA3C,EAAgDD,IAAI,UAApD,EAFF,EAGE,EAAEF,OAAO,GAAT,EAAcC,MAAM,EAApB,EAHF,CAJgB,CAAlB;;AAWe,SAASH,OAAT,CAAiBM,KAAjB,EAAwB;AACrC,SACE;AAAA;AAAA,MAAK,WAAW,0BAAW,iBAAOC,IAAlB,EAAwBD,MAAME,SAA9B,CAAhB;AACGP,cAAUQ,GAAV,CAAc,UAACC,GAAD,EAAMC,MAAN;AAAA,aACb;AAAA;AAAA,UAAK,KAAKA,MAAV,EAAkB,WAAW,iBAAOD,GAApC;AACGA,YAAID,GAAJ,CAAQ,UAACG,GAAD,EAAS;AAChB,cAAIN,MAAMO,WAAN,KAAsBD,IAAIV,KAAJ,KAAc,GAAd,IAAqBU,IAAIV,KAAJ,KAAc,GAAzD,CAAJ,EAAmE;AACjE,mBACE,uCAAK,KAAKU,IAAIV,KAAd,EAAqB,WAAW,iBAAOY,cAAvC,GADF;AAGD;AACD,iBACE;AACE,iBAAKF,IAAIV,KADX;AAEE,iBAAKU,GAFP;AAGE,qBAASN,MAAMS,aAHjB;AAIE,sBAAUT,MAAMU,cAJlB;AAKE,gCAAoBV,MAAMW;AAL5B,YADF;AASD,SAfA;AADH,OADa;AAAA,KAAd;AADH,GADF;AAwBD;;AAEDjB,QAAQkB,SAAR,GAAoB;AAClBV,aAAW,oBAAUW,MADH;AAElBN,eAAa,oBAAUO,IAFL;AAGlBL,iBAAe,oBAAUM,IAHP;AAIlBL,kBAAgB,oBAAUK,IAJR;AAKlBJ,sBAAoB,oBAAUK;AALZ,CAApB","file":"index.js","sourcesContent":["import React from 'react';\nimport PropTypes from 'prop-types';\nimport classnames from 'classnames';\n\nimport DialButton from '../DialButton';\nimport styles from './styles.scss';\n\nconst keyConfig = [\n  [{ value: '1', text: '' }, { value: '2', text: 'ABC', dx: '183.211' }, { value: '3', text: 'DEF', dx: '188.633' }],\n  [{ value: '4', text: 'GHI', dx: '192.242' }, { value: '5', text: 'JKL', dx: '194.0235' }, { value: '6', text: 'MNO', dx: '174.211' }],\n  [{ value: '7', text: 'PQRS', dx: '163.3595' }, { value: '8', text: 'TUV', dx: '188.633' }, { value: '9', text: 'WXYZ', dx: '161.5705' }],\n  [\n    { value: '*', text: '' },\n    { value: '0', text: '+', alternativeValue: '+', dx: '228.5625' },\n    { value: '#', text: '' },\n  ],\n];\n\nexport default function DialPad(props) {\n  return (\n    <div className={classnames(styles.root, props.className)}>\n      {keyConfig.map((row, rowIdx) => (\n        <div key={rowIdx} className={styles.row}>\n          {row.map((btn) => {\n            if (props.hideSpecial && (btn.value === '*' || btn.value === '#')) {\n              return (\n                <div key={btn.value} className={styles.btnPlaceholder} />\n              );\n            }\n            return (\n              <DialButton\n                key={btn.value}\n                btn={btn}\n                onPress={props.onButtonPress}\n                onOutput={props.onButtonOutput}\n                alternativeTimeout={props.alternativeTimeout}\n              />\n            );\n          })}\n        </div>\n      ))}\n    </div>\n  );\n}\n\nDialPad.propTypes = {\n  className: PropTypes.string,\n  hideSpecial: PropTypes.bool,\n  onButtonPress: PropTypes.func,\n  onButtonOutput: PropTypes.func,\n  alternativeTimeout: PropTypes.number,\n};\n"]}