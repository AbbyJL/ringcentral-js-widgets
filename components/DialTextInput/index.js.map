{"version":3,"sources":["components/DialTextInput/index.js"],"names":["DialTextInput","className","invalid","value","onChangeEvent","onDelete","dialInput","to","getString","dial_Input","delete","icon_clear","propTypes","string","bool","func","defaultValue","input"],"mappings":";;;;;;AAAA;;;;AACA;;;;AACA;;;;AACA;;;;AACA;;;;AACA;;;;AACA;;;;;;AAEA,SAASA,aAAT,OAMG;AAAA,MALDC,SAKC,QALDA,SAKC;AAAA,MAJDC,OAIC,QAJDA,OAIC;AAAA,MAHDC,KAGC,QAHDA,KAGC;AAAA,MAFDC,aAEC,QAFDA,aAEC;AAAA,MADDC,QACC,QADDA,QACC;;AACD,SACE;AAAA;AAAA;AACE,iBAAW,0BACT,iBAAOC,SADE,EAETL,SAFS,EAGTC,WAAW,iBAAOA,OAHT,CADb;AAME;AAAA;AAAA,QAAM,WAAW,0BAAW,iBAAOK,EAAlB,CAAjB;AACG,qBAAKC,SAAL,CAAe,IAAf;AADH,KANF;AASE;AACE,iBAAW,iBAAOC,UADpB;AAEE,aAAON,KAFT;AAGE,gBAAUC;AAHZ,MATF;AAcE,4CAAM,WAAW,0BAAW,iBAAOM,MAAlB,EAA0B,uBAAaC,UAAvC,CAAjB,EAAqE,SAASN,QAA9E;AAdF,GADF;AAmBD;;AAEDL,cAAcY,SAAd,GAA0B;AACxBX,aAAW,iBAAUY,MADG;AAExBX,WAAS,iBAAUY,IAFK;AAGxBX,SAAO,iBAAUU,MAHO;AAIxBT,iBAAe,iBAAUW,IAJD;AAKxBV,YAAU,iBAAUU;AALI,CAA1B;AAOAf,cAAcgB,YAAd,GAA6B;AAC3Bf,aAAW,iBAAOgB;AADS,CAA7B;;kBAKejB,a","file":"index.js","sourcesContent":["import React, { Component, PropTypes } from 'react';\nimport classnames from 'classnames';\nimport styles from './styles.scss';\nimport i18n from './i18n';\nimport TextInput from '../TextInput';\nimport rcFont from '../../assets/RcFont/RcFont.scss';\nimport dynamicsFont from '../../assets/DynamicsFont/DynamicsFont.scss';\n\nfunction DialTextInput({\n  className,\n  invalid,\n  value,\n  onChangeEvent,\n  onDelete,\n}) {\n  return (\n    <div\n      className={classnames(\n        styles.dialInput,\n        className,\n        invalid && styles.invalid,\n        )}>\n      <span className={classnames(styles.to)}>\n        {i18n.getString('to')}\n      </span>\n      <TextInput\n        className={styles.dial_Input}\n        value={value}\n        onChange={onChangeEvent}\n      />\n      <span className={classnames(styles.delete, dynamicsFont.icon_clear)} onClick={onDelete}>\n      </span>\n    </div>\n  );\n}\n\nDialTextInput.propTypes = {\n  className: PropTypes.string,\n  invalid: PropTypes.bool,\n  value: PropTypes.string,\n  onChangeEvent: PropTypes.func,\n  onDelete: PropTypes.func,\n};\nDialTextInput.defaultValue = {\n  className: styles.input\n};\n\n\nexport default DialTextInput;\n"]}