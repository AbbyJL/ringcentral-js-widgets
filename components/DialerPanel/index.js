'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _DialPad = require('../DialPad');

var _DialPad2 = _interopRequireDefault(_DialPad);

var _RecipientsInput = require('../RecipientsInput');

var _RecipientsInput2 = _interopRequireDefault(_RecipientsInput);

var _FromField = require('../FromField');

var _FromField2 = _interopRequireDefault(_FromField);

var _SpinnerOverlay = require('../SpinnerOverlay');

var _SpinnerOverlay2 = _interopRequireDefault(_SpinnerOverlay);

var _CircleButton = require('../CircleButton');

var _CircleButton2 = _interopRequireDefault(_CircleButton);

var _Answer = require('../../assets/images/Answer.svg');

var _Answer2 = _interopRequireDefault(_Answer);

var _styles = require('./styles.scss');

var _styles2 = _interopRequireDefault(_styles);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function DialerPanel(_ref) {
  var callButtonDisabled = _ref.callButtonDisabled,
      className = _ref.className,
      onToNumberChange = _ref.onToNumberChange,
      onCallButtonClick = _ref.onCallButtonClick,
      toNumber = _ref.toNumber,
      fromNumber = _ref.fromNumber,
      fromNumbers = _ref.fromNumbers,
      changeFromNumber = _ref.changeFromNumber,
      formatPhone = _ref.formatPhone,
      isWebphoneMode = _ref.isWebphoneMode,
      currentLocale = _ref.currentLocale,
      showSpinner = _ref.showSpinner,
      dialButtonVolume = _ref.dialButtonVolume,
      dialButtonMuted = _ref.dialButtonMuted,
      searchContact = _ref.searchContact,
      searchContactList = _ref.searchContactList,
      recipient = _ref.recipient,
      clearToNumber = _ref.clearToNumber,
      setRecipient = _ref.setRecipient,
      clearRecipient = _ref.clearRecipient,
      phoneTypeRenderer = _ref.phoneTypeRenderer;

  var onCallFunc = function onCallFunc() {
    if (!callButtonDisabled) {
      onCallButtonClick();
    }
  };
  var content = showSpinner ? _react2.default.createElement(_SpinnerOverlay2.default, null) : null;
  return _react2.default.createElement(
    'div',
    { className: (0, _classnames2.default)(_styles2.default.root, className) },
    _react2.default.createElement(_RecipientsInput2.default, {
      value: toNumber,
      onChange: onToNumberChange,
      onClean: clearToNumber,
      recipient: recipient,
      addToRecipients: setRecipient,
      removeFromRecipients: clearRecipient,
      searchContact: searchContact,
      searchContactList: searchContactList,
      formatContactPhone: formatPhone,
      currentLocale: currentLocale,
      phoneTypeRenderer: phoneTypeRenderer,
      titleEnabled: true,
      autoFocus: true
    }),
    _react2.default.createElement(
      'div',
      { className: _styles2.default.inputFields },
      _react2.default.createElement(_FromField2.default, {
        fromNumber: fromNumber,
        fromNumbers: fromNumbers,
        onChange: changeFromNumber,
        formatPhone: formatPhone,
        currentLocale: currentLocale,
        hidden: !isWebphoneMode
      })
    ),
    _react2.default.createElement(
      'div',
      { className: _styles2.default.dialButtons },
      _react2.default.createElement(_DialPad2.default, {
        className: _styles2.default.dialPad,
        onButtonOutput: function onButtonOutput(key) {
          onToNumberChange(toNumber + key);
        },
        dialButtonVolume: dialButtonVolume,
        dialButtonMuted: dialButtonMuted
      }),
      _react2.default.createElement(
        'div',
        { className: (0, _classnames2.default)(_styles2.default.callBtnRow) },
        _react2.default.createElement(
          'div',
          { className: _styles2.default.callBtn },
          _react2.default.createElement(_CircleButton2.default, {
            className: (0, _classnames2.default)(_styles2.default.dialBtn, callButtonDisabled && _styles2.default.disabled),
            onClick: onCallFunc,
            disabled: callButtonDisabled,
            icon: _Answer2.default,
            showBorder: false
          })
        )
      )
    ),
    content
  );
}
DialerPanel.propTypes = {
  className: _propTypes2.default.string,
  onCallButtonClick: _propTypes2.default.func.isRequired,
  callButtonDisabled: _propTypes2.default.bool,
  isWebphoneMode: _propTypes2.default.bool,
  toNumber: _propTypes2.default.string,
  onToNumberChange: _propTypes2.default.func,
  fromNumber: _propTypes2.default.string,
  currentLocale: _propTypes2.default.string.isRequired,
  fromNumbers: _propTypes2.default.arrayOf(_propTypes2.default.shape({
    phoneNumber: _propTypes2.default.string,
    usageType: _propTypes2.default.string
  })),
  changeFromNumber: _propTypes2.default.func,
  formatPhone: _propTypes2.default.func,
  showSpinner: _propTypes2.default.bool,
  dialButtonVolume: _propTypes2.default.number,
  dialButtonMuted: _propTypes2.default.bool,
  searchContact: _propTypes2.default.func.isRequired,
  searchContactList: _propTypes2.default.arrayOf(_propTypes2.default.shape({
    name: _propTypes2.default.string.isRequired,
    entityType: _propTypes2.default.string.isRequired,
    phoneType: _propTypes2.default.string.isRequired,
    phoneNumber: _propTypes2.default.string.isRequired
  })).isRequired,
  recipient: _propTypes2.default.shape({
    phoneNumber: _propTypes2.default.string.isRequired,
    name: _propTypes2.default.string
  }),
  clearToNumber: _propTypes2.default.func.isRequired,
  setRecipient: _propTypes2.default.func.isRequired,
  clearRecipient: _propTypes2.default.func.isRequired,
  phoneTypeRenderer: _propTypes2.default.func
};

DialerPanel.defaultProps = {
  className: null,
  fromNumber: null,
  callButtonDisabled: false,
  toNumber: '',
  fromNumbers: [],
  isWebphoneMode: false,
  changeFromNumber: function changeFromNumber() {
    return null;
  },
  onToNumberChange: function onToNumberChange() {
    return null;
  },
  formatPhone: function formatPhone(phoneNumber) {
    return phoneNumber;
  },
  showSpinner: false,
  dialButtonVolume: 1,
  dialButtonMuted: false,
  recipient: [],
  phoneTypeRenderer: undefined
};

exports.default = DialerPanel;
//# sourceMappingURL=index.js.map
