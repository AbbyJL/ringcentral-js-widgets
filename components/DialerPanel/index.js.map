{"version":3,"sources":["components/DialerPanel/index.js"],"names":["DialerPanel","callButtonDisabled","className","keepToNumber","onCall","toNumber","fromNumber","fromNumbers","changeFromNumber","formatPhone","isWebphoneMode","currentLocale","showSpinner","onCallFunc","content","root","inputFields","event","currentTarget","value","dialButtons","dialPad","key","callBtnRow","callBtn","dialBtn","disabled","propTypes","string","func","isRequired","bool","arrayOf","shape","phoneNumber","usageType","defaultProps"],"mappings":";;;;;;AAAA;;;;AACA;;;;AACA;;;;AACA;;;;AACA;;;;AACA;;;;AACA;;;;AACA;;;;AACA;;;;AAEA;;;;;;AAEA,SAASA,WAAT,OAaG;AAAA,MAZDC,kBAYC,QAZDA,kBAYC;AAAA,MAXDC,SAWC,QAXDA,SAWC;AAAA,MAVDC,YAUC,QAVDA,YAUC;AAAA,MATDC,MASC,QATDA,MASC;AAAA,MARDC,QAQC,QARDA,QAQC;AAAA,MAPDC,UAOC,QAPDA,UAOC;AAAA,MANDC,WAMC,QANDA,WAMC;AAAA,MALDC,gBAKC,QALDA,gBAKC;AAAA,MAJDC,WAIC,QAJDA,WAIC;AAAA,MAHDC,cAGC,QAHDA,cAGC;AAAA,MAFDC,aAEC,QAFDA,aAEC;AAAA,MADDC,WACC,QADDA,WACC;;AACD,MAAMC,aAAa,SAAbA,UAAa,GAAM;AACvB,QAAI,CAACZ,kBAAL,EAAyB;AACvBG;AACD;AACF,GAJD;AAKA,MAAMU,UAAUF,cAAe,6DAAf,GAAqC,IAArD;AACA,SACE;AAAA;AAAA,MAAK,WAAW,0BAAW,iBAAOG,IAAlB,EAAwBb,SAAxB,CAAhB;AACE;AAAA;AAAA,QAAK,WAAW,iBAAOc,WAAvB;AACE;AACE,eAAOX,QADT;AAEE,uBAAe,uBAACY,KAAD,EAAW;AACxBd,uBAAac,MAAMC,aAAN,CAAoBC,KAAjC;AACD,SAJH;AAKE,kBAAU,oBAAM;AACdhB,uBAAa,EAAb;AACD;AAPH,QADF;AAUE;AACE,oBAAYG,UADd;AAEE,qBAAaC,WAFf;AAGE,kBAAUC,gBAHZ;AAIE,qBAAaC,WAJf;AAKE,uBAAeE,aALjB;AAME,gBAAQ,CAACD;AANX;AAVF,KADF;AAoBE;AAAA;AAAA,QAAK,WAAW,iBAAOU,WAAvB;AACE;AACE,mBAAW,iBAAOC,OADpB;AAEE,wBAAgB,wBAACC,GAAD,EAAS;AACvBnB,uBAAaE,WAAWiB,GAAxB;AACD;AAJH,QADF;AAOE;AAAA;AAAA,UAAK,WAAW,0BAAW,iBAAOC,UAAlB,CAAhB;AACE;AAAA;AAAA,YAAK,WAAW,iBAAOC,OAAvB;AACE;AACE,uBAAW,0BACT,iBAAOC,OADE,EAETxB,sBAAsB,iBAAOyB,QAFpB,CADb;AAKE,qBAASb,UALX;AAME,sBAAUZ,kBANZ;AAOE,kCAPF;AAQE,wBAAY;AARd;AADF;AADF;AAPF,KApBF;AA0CGa;AA1CH,GADF;AA8CD;AACDd,YAAY2B,SAAZ,GAAwB;AACtBzB,aAAW,oBAAU0B,MADC;AAEtBxB,UAAQ,oBAAUyB,IAAV,CAAeC,UAFD;AAGtB7B,sBAAoB,oBAAU8B,IAHR;AAItBrB,kBAAgB,oBAAUqB,IAJJ;AAKtB1B,YAAU,oBAAUuB,MALE;AAMtBzB,gBAAc,oBAAU0B,IANF;AAOtBvB,cAAY,oBAAUsB,MAPA;AAQtBjB,iBAAe,oBAAUiB,MAAV,CAAiBE,UARV;AAStBvB,eAAa,oBAAUyB,OAAV,CAAkB,oBAAUC,KAAV,CAAgB;AAC7CC,iBAAa,oBAAUN,MADsB;AAE7CO,eAAW,oBAAUP;AAFwB,GAAhB,CAAlB,CATS;AAatBpB,oBAAkB,oBAAUqB,IAbN;AActBpB,eAAa,oBAAUoB,IAdD;AAetBjB,eAAa,oBAAUmB;AAfD,CAAxB;;AAkBA/B,YAAYoC,YAAZ,GAA2B;AACzBlC,aAAW,IADc;AAEzBI,cAAY,IAFa;AAGzBL,sBAAoB,KAHK;AAIzBI,YAAU,EAJe;AAKzBE,eAAa,EALY;AAMzBG,kBAAgB,KANS;AAOzBF,oBAAkB;AAAA,WAAM,IAAN;AAAA,GAPO;AAQzBL,gBAAc;AAAA,WAAM,IAAN;AAAA,GARW;AASzBM,eAAa;AAAA,WAAeyB,WAAf;AAAA,GATY;AAUzBtB,eAAa;AAVY,CAA3B;;kBAaeZ,W","file":"index.js","sourcesContent":["import React from 'react';\nimport PropTypes from 'prop-types';\nimport classnames from 'classnames';\nimport DialPad from '../DialPad';\nimport DialTextInput from '../DialTextInput';\nimport CallIdSelect from '../CallIdSelect';\nimport SpinnerOverlay from '../SpinnerOverlay';\nimport CircleButton from '../CircleButton';\nimport AnswerIcon from '../../assets/images/Answer.svg';\n\nimport styles from './styles.scss';\n\nfunction DialerPanel({\n  callButtonDisabled,\n  className,\n  keepToNumber,\n  onCall,\n  toNumber,\n  fromNumber,\n  fromNumbers,\n  changeFromNumber,\n  formatPhone,\n  isWebphoneMode,\n  currentLocale,\n  showSpinner,\n}) {\n  const onCallFunc = () => {\n    if (!callButtonDisabled) {\n      onCall();\n    }\n  };\n  const content = showSpinner ? (<SpinnerOverlay />) : null;\n  return (\n    <div className={classnames(styles.root, className)}>\n      <div className={styles.inputFields}>\n        <DialTextInput\n          value={toNumber}\n          onChangeEvent={(event) => {\n            keepToNumber(event.currentTarget.value);\n          }}\n          onDelete={() => {\n            keepToNumber('');\n          }}\n          />\n        <CallIdSelect\n          fromNumber={fromNumber}\n          fromNumbers={fromNumbers}\n          onChange={changeFromNumber}\n          formatPhone={formatPhone}\n          currentLocale={currentLocale}\n          hidden={!isWebphoneMode}\n        />\n      </div>\n      <div className={styles.dialButtons}>\n        <DialPad\n          className={styles.dialPad}\n          onButtonOutput={(key) => {\n            keepToNumber(toNumber + key);\n          }}\n          />\n        <div className={classnames(styles.callBtnRow)}>\n          <div className={styles.callBtn}>\n            <CircleButton\n              className={classnames(\n                styles.dialBtn,\n                callButtonDisabled && styles.disabled,\n              )}\n              onClick={onCallFunc}\n              disabled={callButtonDisabled}\n              icon={AnswerIcon}\n              showBorder={false}\n            />\n          </div>\n        </div>\n      </div>\n      {content}\n    </div>\n  );\n}\nDialerPanel.propTypes = {\n  className: PropTypes.string,\n  onCall: PropTypes.func.isRequired,\n  callButtonDisabled: PropTypes.bool,\n  isWebphoneMode: PropTypes.bool,\n  toNumber: PropTypes.string,\n  keepToNumber: PropTypes.func,\n  fromNumber: PropTypes.string,\n  currentLocale: PropTypes.string.isRequired,\n  fromNumbers: PropTypes.arrayOf(PropTypes.shape({\n    phoneNumber: PropTypes.string,\n    usageType: PropTypes.string,\n  })),\n  changeFromNumber: PropTypes.func,\n  formatPhone: PropTypes.func,\n  showSpinner: PropTypes.bool,\n};\n\nDialerPanel.defaultProps = {\n  className: null,\n  fromNumber: null,\n  callButtonDisabled: false,\n  toNumber: '',\n  fromNumbers: [],\n  isWebphoneMode: false,\n  changeFromNumber: () => null,\n  keepToNumber: () => null,\n  formatPhone: phoneNumber => phoneNumber,\n  showSpinner: false,\n};\n\nexport default DialerPanel;\n"]}