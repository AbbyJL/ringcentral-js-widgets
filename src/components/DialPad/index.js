import React, { Component, PropTypes } from 'react';
import classnames from 'classnames';
import styles from './styles.scss';
import audios from './audios';

const keyConfig = [
  [{ value: '1', text: '' }, { value: '2', text: 'ABC' }, { value: '3', text: 'DEF' }],
  [{ value: '4', text: 'GHI' }, { value: '5', text: 'JKL' }, { value: '6', text: 'MON' }],
  [{ value: '7', text: 'PQRS' }, { value: '8', text: 'TUV' }, { value: '9', text: 'WXYZ' }],
  [
    { value: '*', text: '' },
    { value: '0', text: '+', alternativeValue: '+' },
    { value: '#', text: '' },
  ],
];

const ALTERNATIVE_TIMEOUT = 1000;

class DialButton extends Component {
  constructor(props, ...args) {
    super(props, ...args);
    this.state = {
      pressed: false,
    };
    this.timeout = null;
    if (typeof document !== 'undefined' && document.createElement && audios[props.btn.value]) {
      this.audio = document.createElement('audio');
      this.audio.src = audios[props.btn.value];
    }
    this.onMouseDown = () => {
      if (this.audio && this.audio.canPlayType('audio/ogg') !== '') {
        this.audio.currentTime = 0;
        this.audio.play();
      }
      if (typeof this.props.onPress === 'function') {
        this.props.onPress(this.props.btn.value);
      }

      this.timeout = setTimeout(() => {
        if (this.state.pressed) {
          if (this.timeout) {
            clearTimeout(this.timeout);
            this.timeout = null;
          }
          if (typeof this.props.onOutput === 'function') {
            this.props.onOutput(this.props.btn.alternativeValue || this.props.btn.value);
          }
          this.setState({
            pressed: false,
          });
        }
      }, this.props.alternativeTimeout || ALTERNATIVE_TIMEOUT);

      this.setState({
        pressed: true,
      });
    };
    this.onMouseUp = () => {
      if (this.state.pressed) {
        if (this.timeout) {
          clearTimeout(this.timeout);
          this.timeout = null;
        }
        if (typeof this.props.onOutput === 'function') {
          this.props.onOutput(this.props.btn.value);
        }
        this.setState({
          pressed: false,
        });
      }
    };
    this.onMouseLeave = () => {
      if (this.state.pressed) {
        if (this.timeout) {
          clearTimeout(this.timeout);
          this.timeout = null;
        }
        this.setState({
          pressed: false,
        });
      }
    };
  }
  render() {
    return (
      <div className={styles.btnPlaceholder}>
        <svg className={styles.btnSvg} viewBox="0 0 500 500">
          <g
            className={classnames(
              styles.btnSvgGroup,
              this.state.pressed && styles.pressed
            )}
            onMouseUp={this.onMouseUp}
            onMouseDown={this.onMouseDown}
            onMouseLeave={this.onMouseLeave}>
            <circle
              className={styles.circle}
              cx="250"
              cy="250"
              r="200"
            />
            <text
              className={styles.btnValue}
              x="250"
              y="280">
              {this.props.btn.value}
            </text>
            <text
              className={styles.btnText}
              x="250"
              y="380">
              {this.props.btn.text}
            </text>
          </g>
        </svg>
      </div>
    );
  }
}

DialButton.propTypes = {
  btn: PropTypes.shape({
    value: PropTypes.string.isRequired,
    text: PropTypes.string,
    alternativeValue: PropTypes.string,
  }).isRequired,
  audio: PropTypes.string,
  onPress: PropTypes.func,
  onOutput: PropTypes.func,
  alternativeTimeout: PropTypes.number,
};

export default function DialPad(props) {
  return (
    <div className={classnames(styles.root, props.className)}>
      {keyConfig.map((row, rowIdx) => (
        <div key={rowIdx} className={styles.row}>
          {row.map(btn => {
            if (props.hideSpecial && (btn.value === '*' || btn.value === '#')) {
              return (
                <div key={btn.value} className={styles.btnPlaceholder} />
              );
            }
            return (
              <DialButton
                key={btn.value}
                btn={btn}
                onPress={props.onButtonPress}
                onOutput={props.onButtonOutput}
                alternativeTimeout={props.alternativeTimeout}
              />
            );
          })}
        </div>
      ))}
    </div>
  );
}

DialPad.propTypes = {
  className: PropTypes.string,
  hideSpecial: PropTypes.bool,
  onButtonPress: PropTypes.func,
  onButtonOutput: PropTypes.func,
  alternativeTimeout: PropTypes.number,
};
