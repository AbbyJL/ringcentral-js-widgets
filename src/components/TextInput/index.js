import React, { Component, PropTypes } from 'react';
import classnames from 'classnames';
import styles from './styles.scss';

class TextInput extends Component {
  constructor(props){
    super(props);
    this.state = {
      value: (props.value) ? props.value : ''
    };
  }

  render() {
    const props = this.props;
    // console.debug('class',props.className);
    return (
      <div
        className={classnames(
          styles.root,
          props.className,
          props.invalid && styles.invalid,
          )}>
        <input
          onChange={props.onChange}
          placeholder={props.placeholder}
          disabled={props.disabled}
          readOnly={props.readOnly}
          pattern={props.pattern}
          maxLength={props.maxLength}
          name={props.name}
          value={props.value}
          defaultValue={props.defaultValue}
          className={classnames(styles.input)}
          onKeyDown={props.onKeyDown}
        />
      </div>
    );
  }
}

TextInput.propTypes = {
  className: PropTypes.string,
  onChange: PropTypes.func,
  placeholder: PropTypes.string,
  disabled: PropTypes.bool,
  readOnly: PropTypes.bool,
  pattern: PropTypes.string,
  maxLength: PropTypes.number,
  name: PropTypes.string,
  value: PropTypes.string,
  defaultValue: PropTypes.string,
  invalid: PropTypes.bool,
  onKeyDown: PropTypes.func,
};
TextInput.defaultValue = {
  className: styles.input
};


export default TextInput;
