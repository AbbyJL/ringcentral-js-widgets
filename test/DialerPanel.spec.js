import { Provider } from 'react-redux';
import { Router } from 'react-router';
import toJson from 'enzyme-to-json';

import { getWrapper, getState } from './shared';
import DialerPanel from '../src/components/DialerPanel';
import DialTextInput from '../src/components/DialTextInput';
import TextInput from '../src/components/TextInput';
import DialPad, { DialButton } from '../src/components/DialPad';

let panel = null;
let wrapper = null;
beforeEach(() => {
  wrapper = getWrapper();
  panel = wrapper.find(Provider).first()
    .find(Router).first()
    .find(DialerPanel)
    .first();
});

const clickButton = (button) => {
  button.find('g').first().props().onMouseDown();
  button.find('g').first().props().onMouseUp();
};

describe('dialer panel', () => {
  test('dial text input', () => {
    const dialTextInput = panel.find(DialTextInput).first();
    expect(dialTextInput).toBeDefined();
    const textInput = dialTextInput.find(TextInput).first();
    expect(textInput).toBeDefined();
    expect(textInput.props().value).toEqual('');

    textInput.props().onChange({ currentTarget: { value: '16506417422' } });
    expect(textInput.props().value).toEqual('16506417422');
  });

  test('dial buttons', () => {
    const dialPad = panel.find(DialPad).first();
    const buttons = dialPad.find(DialButton);
    expect(buttons.length).toEqual(12);
    const button1 = buttons.at(0);
    expect(button1.props().btn.value).toEqual('1');
    const button2 = buttons.at(1);
    expect(button2.props().btn.value).toEqual('2');
    const button3 = buttons.at(2);
    expect(button3.props().btn.value).toEqual('3');
    const button4 = buttons.at(3);
    expect(button4.props().btn.value).toEqual('4');
    const button5 = buttons.at(4);
    expect(button5.props().btn.value).toEqual('5');
    const button6 = buttons.at(5);
    expect(button6.props().btn.value).toEqual('6');
    const button7 = buttons.at(6);
    expect(button7.props().btn.value).toEqual('7');
    const button8 = buttons.at(7);
    expect(button8.props().btn.value).toEqual('8');
    const button9 = buttons.at(8);
    expect(button9.props().btn.value).toEqual('9');
    const buttonStar = buttons.at(9);
    expect(buttonStar.props().btn.value).toEqual('*');
    const button0 = buttons.at(10);
    expect(button0.props().btn.value).toEqual('0');
    const buttonSharp = buttons.at(11);
    expect(buttonSharp.props().btn.value).toEqual('#');

    const textInput = panel.find(DialTextInput).first().find(TextInput).first();
    expect(textInput.props().value).toEqual('');
    clickButton(button0);
    clickButton(button1);
    clickButton(button2);
    clickButton(button3);
    clickButton(button4);
    clickButton(button5);
    clickButton(button6);
    clickButton(button7);
    clickButton(button8);
    clickButton(button9);
    clickButton(buttonStar);
    clickButton(buttonSharp);
    expect(textInput.props().value).toEqual('0123456789*#');

    expect(getState(wrapper).call.toNumber).toEqual('0123456789*#');

    expect(toJson(textInput)).toMatchSnapshot();
  });
});
