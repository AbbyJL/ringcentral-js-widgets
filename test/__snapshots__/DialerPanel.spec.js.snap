// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dialer panel dial buttons 1`] = `
<TextInput
  disabled={false}
  invalid={false}
  onChange={[Function]}
  readOnly={false}
  value="0123456789*#"
>
  <div
    className=""
  >
    <input
      disabled={false}
      onChange={[Function]}
      readOnly={false}
      value="0123456789*#"
    />
  </div>
</TextInput>
`;
